src/actix/api/update_api.rs

```diff
@@ -430,6 +430,70 @@ async fn delete_field_index(
     process_response(response, timing, None)
 }
 
+/// Request body for the staging test delay endpoint.
+/// Only available when the `staging` feature is enabled.
+#[cfg(feature = "staging")]
+#[derive(Debug, Deserialize, Validate)]
+pub struct TestDelayRequest {
+    /// Duration of the delay in seconds (default: 1.0, max: 300.0).
+    #[serde(default = "default_test_delay_duration")]
+    #[validate(range(min = 0.0, max = 300.0))]
+    pub duration: f64,
+}
+
+#[cfg(feature = "staging")]
+fn default_test_delay_duration() -> f64 {
+    1.0
+}
+
+/// Staging endpoint that introduces an artificial delay for testing purposes.
+/// Only available when the `staging` feature is enabled.
+#[cfg(feature = "staging")]
+#[post("/collections/{name}/points/staging")]
+async fn staging_test_delay(
+    dispatcher: web::Data<Dispatcher>,
+    collection: Path<CollectionPath>,
+    operation: Json<TestDelayRequest>,
```

Comment:
I'd be better to accept a PointOperation type here. Or a dedicated type for staging stuff once we've added a variant in `CollectionUpdateOperations`.

It'll be a bit more verbose in usage, but it also means all new operations we add are immediately usable without implementing a new API.